client/src/graphs/templates/helpers.ts

```diff
@@ -0,0 +1,68 @@
+import type { GNode } from "@graph/types";
+import type { BoundingBox, Coordinate } from "@shape/types";
+
+export const getAverageCoordinatesOfGraphNodes = (nodes: GNode[]): Coordinate => {
+  if (nodes.length === 0) {
+    return { x: 0, y: 0 };
+  }
+
+  const total = nodes.reduce((acc, node) => {
+    acc.x += node.x;
+    acc.y += node.y;
+    return acc;
+  }, { x: 0, y: 0 });
+
+  const nodeCount = nodes.length;
+  return {
+    x: total.x / nodeCount,
+    y: total.y / nodeCount
+  };
+};
+
+export const getGraphNodesBoundingBox = (nodes: GNode[], padding: number = 0): BoundingBox => {
```

Comment:
this is already done for you in Marquee updateEncapsulatedNodeBox. move your tests to the marquee plugin just as it appears in graph/base (take a look). and then take the function for updateEncapsulatedNodeBox and move it into its adjacent files and then import back to maqree and use it here